### DISCUSSION TITLE
validation against satl data

### DISCUSSION THREAD

**Discussion by ghost** [dtcenter/METplus#1813] [Configuration] [answered]
Hi,
I am trying to use satellite data (which is at every 3 hours and covers part of the globe) to validate against model forecast. As satellite coverage is only partial, I would like to generate statistics (GridStat) not for every 3 hours, but accumulate it over day1, day2.. format. Or whatever meaningful way you can suggest. Can you please help me on this. 
You may look into script in here /lfs/h2/emc/vpppg/noscrub/Partha.Bhattacharjee/Aerosol/METPlus/Scripts/grid_stat_VIIRS_Taod.conf

Any help is appreciated.

**Comment by JohnHalleyGotway** [comment]
Hi @perthsb, I see you have a question about how to aggregate satellite vx results. I have some suggestions to offer.

First, I assume that you do NOT want to aggregate the raw satellite data over a 1 or 2 day period. That'd be like taking the union of many satellite passes over multiple hours/days and placing them into a single field of data. That'd result in an analysis field with fewer missing data points, but the data points would have very different timestamps. And that's challenging because I assume you want to compare model output to satellite observations for the same (or at least pretty similar) timestamps.

Instead, I'd recommend processing each 3 hour satellite observation file against all forecasts that are valid at that time (or close to it). You'd just run Grid-Stat to compare the 3 hour satellite data to the model output. And when you do, just set `mask = { grid = [ "FULL"]; }}`. That'll use the *full* model domain, but will only include grid points for which both forecast and observation data are present. For one time, that'll be one piece of the globe. For 3 hours later, that'll be another piece of the globe... as time passes, the location of the orbital pass changes as well.

Be sure to configure Grid-Stat to write out contingency tables (CTC and/or MCTC) and/or partial sums (SL1L2).

And once you have some output, just run Stat-Analysis to aggregate that "FULL" data over one or more days. For example:
```
stat_analysis -lookin grid_stat_output_directory \
-job aggregate_stat -line_type SL1L2 -out_line_type CNT \
-fcst_valid_beg 20220903_00 -fcst_valid_end 20220904_00 \
-by FCST_VAR -vx_mask FULL \
-out_stat sl1l2_to_cnt.stat
```
That'll aggregate all the SL1L2 lines for the "FULL" domain falling between those timestamps into a single set of SL1L2 partial sums. And then it'll derive corresponding continuous statistics. Here's a similar command for handle CTC lines if you're doing categorical verification:
```
stat_analysis -lookin grid_stat_output_directory \
-job aggregate_stat -line_type CTC -out_line_type CTS \
-fcst_valid_beg 20220903_00 -fcst_valid_end 20220904_00 \
-by FCST_VAR -by FCST_THRESH -vx_mask FULL \
-out_stat ctc_to_cts.stat
```
I can't actually access WCOSS2 right now, so I can't directly test with your data. But hopefully this approach makes sense.

Is that what you were thinking or do you have something else in mind?

**Reply by ghost** [reply]
Thanks @JohnHalleyGotway .I brought the data to Hera 
/scratch1/NCEPDEV/stmp2/Partha.Bhattacharjee/METplus/Scripts/Scripts/run_metplus_viirsTaod.sh 
Also attached a figure that shows coverage of satellite that the model forecast will be validating against. 
![coverage](https://user-images.githubusercontent.com/88152718/191349778-0a1a7151-4da3-4783-8a9f-55d9f19e98e7.png)

**Reply by JohnHalleyGotway** [reply]
@perthsb I haven't had a chance to take a look at your sample data. I do think that using `mask.grid = [ "FULL" ];` and then aggregating results across a full day would have the desired effect.

Please let me know if there something specific you want me to look at and address.

**Reply by ghost** [reply]
Hi @JohnHalleyGotway , I have run SeriesAnalysis after running GridStat in Hera and getting the follwoing error : 
 ERROR: (stat_analysis_wrapper.py:271) Field information defined in both [FCST/OBS]_VAR_LIST and [FCST/OBS]_VAR<n>_[NAME/LEVELS]. Use one or the other formats to run

I run this configure file for StatAnlysis : /scratch1/NCEPDEV/stmp2/Partha.Bhattacharjee/METplus/Scripts/Scripts/StatAnlysis_VIIRS_Taod.conf after running GridStat using
/scratch1/NCEPDEV/stmp2/Partha.Bhattacharjee/METplus/Scripts/Scripts/grid_stat_VIIRS_Taod.conf

Any help on what needs to be changed in configure file . Thanks

**Reply by JohnHalleyGotway** [reply]
Hi @perthsb sorry for the long delay. I was able to log on to hera after it came back up this afternoon and take a look.

I'm certainly able to run Stat-Analysis directly on the command line on your data, as shown below:
```
stat_analysis -lookin /scratch1/NCEPDEV/stmp2/Partha.Bhattacharjee/METplus/Scripts/Scripts/output/Gridstat/grid_stat -job aggregate_stat -line_type SL1L2 -out_line_type CNT -by FCST_LEAD -out_stat sl1l2_to_cnt.txt
```
That generates CNT stats for each of 8 different lead times (0, 3, 6, 9, 12, 15, 18, and 21). Looks like you have 2 output data for each lead time. Here's the output file: 
[sl1l2_to_cnt.txt](https://github.com/dtcenter/METplus/files/9710960/sl1l2_to_cnt.txt)

So the question really is, how do you run that job through METplus?

@georgemccabe are you able to provide @perthsb some guidance on how he should be configuring calls to Stat-Analysis via METplus? I know you've been working on the wrapper recently and figure you'd be the best person to advise.

**Comment by georgemccabe** [comment, marked as answer]
The error message notes that you cannot set field information using both [FCST/OBS]_VAR_LIST and [FCST/OBS]\_VAR\_[NAME/LEVELS]. Remove the values set for FCST_VAR_LIST, FCST_VAR_LEVELS, OBS_VAR_LIST, and OBS_VAR_LEVELS since you have the field information already defined using [FCST/OBS]_VAR1_[NAME/LEVELS].

**Reply by georgemccabe** [reply]
Hi @perthsb , I am just following up on this discussion. Do you need any more assistance?

**Reply by georgemccabe** [reply]
Hi @perthsb, I am going to mark this discussion as answered and lock it to prevent future posts. If you need further assistance on this issue, please create a new discussion topic and reference this discussion's number. Thanks.
